src/test/java/roomescape/domain/ReservationTimeTest.java

```diff
@@ -0,0 +1,18 @@
+package roomescape.domain;
+
+import java.util.function.Supplier;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class ReservationTimeTest {
+    @Test
+    void validate() {
+        // given
+        Supplier<ReservationTime> supplier = () -> new ReservationTime(1L, null);
+
+        // when & then
+        Assertions.assertThatNullPointerException().isThrownBy(
+                supplier::get
+        );
+    }
```

Comment:
Supplier 사용은 `given / when & then` 구분 짓기 위함인가요?? 아래와 같이 합치지 않은 이유가 궁금해요~

```suggestion
    void validate() {
        Assertions.assertThatNullPointerException().isThrownBy(
                new ReservationTime(1L, null);
        );
    }
```

Reply:
네. 생성 자체가 테스트 로직일 경우 `given & when & then` 이 모두 합쳐져 있는 게 복잡해보여서
`given`과 `when & then`을 분리해봤어요.

사실 비교해서 보니까... 기존 방식에서 더 간결해졌다는 느낌이 안 들긴 하네요..

Reply:
히스타는 `given & when & then`의 각 단계가 무엇을 요구한다고 생각하시나요? 지금의 `given`은 어떠한 의미를 담고 있나요? 저는 `given`이 테스트에서 꽤나 중요한 요소라고 생각하는데요. `given`에 따라 테스트가 성공할 수도 실패할 수 있어요. 지금의 `validate()` 테스트에서는 검증하고자 하는 요소인 startAt 파라미터의 null 여부겠지요. 따라서 저라면 아래와 같이 테스트를 작성했을 듯 해요. 그치만 이런 방식은 취향에 따라 갈리기 때문에, 히스타가 편한 방식으로 해주시면 돼요~ㅎ

```suggestion
    void validate() {
        // given
        LocalTime startAt = null;

        // when & then
        assertThatNullPointerException()
                .isThrownBy(() -> new ReservationTime(1L, startAt));
    }```